### PR TITLE
fix: skips submodule metdata autogen if readme isn't present

### PR DESCRIPTION
**Pull request by g-awmalik**

### CODE REVIEW

File: cli/bpmetadata/cmd.go [hunk 1]
```diff
@@ -77,8 +77,22 @@ func generate(cmd *cobra.Command, args []string) error {
...
-	for _, path := range allBpPaths {
-		err = generateMetadataForBpPath(path)
+	for _, modPath := range allBpPaths {
+		// check if module path has readme.md
+		_, err := os.Stat(path.Join(modPath, readmeFileName))
+
+		// throw an error and exit if root module doesn't have a readme.md
+		if !strings.Contains(modPath, modulesPath+"/") && err != nil {
+			return fmt.Errorf("Top-level module does not have a readme. Details: %w\n", err)
+		}
```
Comment (bharathkkb) [inline cli/bpmetadata/cmd.go:87]: It seems like if we move this check to after line 59 we can fail fast and potentially simplify the logic?

File: cli/bpmetadata/cmd.go [hunk 2]
```diff
@@ -77,8 +77,22 @@ func generate(cmd *cobra.Command, args []string) error {
...
+
+		// log info if a sub-module doesn't have a readme.md and continue
+		if err != nil {
+			Log.Info("Skiping metadata for sub-module", "Path:", modPath)
```
Comment (bharathkkb) [inline cli/bpmetadata/cmd.go:91]: So users are aware of why
```suggestion
			Log.Info("Skipping metadata for sub-module identified as an internal module", "Path:", modPath)
```

**Review by bharathkkb — APPROVED** [review]
Thanks @g-awmalik 
Would be good to add a test

File: cli/bpmetadata/cmd.go (hunk 2 above)
Comment (g-awmalik) [reply cli/bpmetadata/cmd.go:91]: done

File: cli/bpmetadata/cmd.go (hunk 1 above)
Comment (g-awmalik) [reply cli/bpmetadata/cmd.go:87]: done.

**Pull request merged**